### PR TITLE
User Service

### PR DESCRIPTION
**Pull request by hrishikeshpaul**
Added basic user service.

### Functionality
1. API to log the user in.
    - API route: `http://localhost:5000/users/login`
    - Type: POST
    - Req body: `name: string, email: string, photo: string`
    - Retrieves the user from the database based on the email address. Returns 200 if the user is there.
    - If that email is not there, then adds the user to the database.Returns a 201 after adding and the user object.
    - Returns 400 if it can't be added or there is an error.

### CODE REVIEW

File: user-service/config/database.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const mongoose = require("mongoose");
+const uri = "mongodb://localhost:27017/scrapbookUserService"
```
Comment (jbhushan791) [inline user-service/config/database.js:2]: is it possible to rename database name like remove service from it or so ?

File: user-service/models/users.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+var mongoose = require('mongoose')
```
Comment (jbhushan791) [inline user-service/models/users.js:1]: Entity is always singular. Change from users to user

**Comment by jbhushan791** [comment]
Also for nomenclature, I would recommend to follow this. This is a common practice.

scrapbook-<someName>-service.

**Review by chirag-indi — APPROVED** [review]

File: user-service/config/database.js (hunk 1 above)
Comment (hrishikeshpaul) [reply user-service/config/database.js:2]: What do you want the name to be? `scrapbookUser`? 

I think it'll be good to keep the word service too. Like srapbookImageService etc. Let me know what you think!

File: user-service/models/users.js (hunk 2 above)
Comment (hrishikeshpaul) [reply user-service/models/users.js:1]: Yes will change that

**Comment by hrishikeshpaul** [comment]
@jbhushan791 alright, will update it!

File: user-service/config/database.js (hunk 1 above)
Comment (jbhushan791) [reply user-service/config/database.js]: Ok, keeping service is also fine.

**Review by jbhushan791 — APPROVED** [review]

**Pull request merged**